data/transactions/logic/eval.go

```diff
@@ -2600,34 +2608,34 @@ func opItxnas(cx *EvalContext) error {
 	return nil
 }
 
-func (cx *EvalContext) getLastInner() []transactions.SignedTxnWithAD {
+func (cx *EvalContext) getLastInner() ([]transactions.SignedTxnWithAD, int) {
 	inners := cx.txn.EvalDelta.InnerTxns
 	// If there are no inners yet, return empty slice, which will result in error
 	if len(inners) == 0 {
-		return inners
+		return inners, 0
 	}
-	return inners[len(inners)-1:]
+	return inners[len(inners)-1:], len(inners) - 1
 }
 
-func (cx *EvalContext) getLastInnerGroup() []transactions.SignedTxnWithAD {
+func (cx *EvalContext) getLastInnerGroup() ([]transactions.SignedTxnWithAD, int) {
 	inners := cx.txn.EvalDelta.InnerTxns
 	// If there are no inners yet, return empty slice, which will result in error
 	if len(inners) == 0 {
-		return inners
+		return inners, 0
 	}
 	gid := inners[len(inners)-1].Txn.Group
 	// If last inner was a singleton, return it as a slice.
 	if gid.IsZero() {
-		return inners[len(inners)-1:]
+		return inners[len(inners)-1:], len(inners) - 1
 	}
 	// Look back for the first non-matching inner (by group) to find beginning
 	for i := len(inners) - 2; i >= 0; i-- {
 		if inners[i].Txn.Group != gid {
-			return inners[i+1:]
+			return inners[i+1:], i + 1
 		}
 	}
 	// All have the same (non-zero) group. Return all
-	return inners
+	return inners, 0
```

Comment:
doubt this is right

Reply:
I think this is right, but should be better explained.  You're trying to let the caller know where the returned slice started, so it can be used as an offset in the hashid, right?

Reply:
Maybe we should refactor this to `getLastInnerGroupStart` and only return that int.  Let the caller do the slicing.  That would make it clear what that int it.